roofit/codegen/src/CodegenImpl.cxx

```diff
@@ -241,6 +243,58 @@ void codegenImpl(RooMultiVarGaussian &arg, CodegenContext &ctx)
                  ctx.buildCall(mathFunc("multiVarGaussian"), arg.xVec().size(), arg.xVec(), arg.muVec(), covISpan));
 }
 
+void codegenImpl(RooMultiPdf &arg, CodegenContext &ctx)
+{
+   int numPdfs = arg.getNumPdfs();
+
+   // MathFunc call
+
+   if (numPdfs > 2) { // the value of this number should be discussed.Beyound a certain number of indices MathFunc call
+                      // becomes more efficient.
+      ctx.addResult(&arg, ctx.buildCall(mathFunc("multipdf"), arg.indexCategory(), arg.getPdfList()));
+
+      std::cout << "MathFunc call used\n";
+
+   } else {
+
+      // Ternary nested expression
+      std::string indexExpr = ctx.getResult(arg.indexCategory());
+
+      // int numPdfs = arg.getNumPdfs();
+      std::string expr;
+
+      for (int i = 0; i < numPdfs; ++i) {
+         RooAbsPdf *pdf = arg.getPdf(i);
+         std::string pdfExpr = ctx.getResult(*pdf);
+
+         expr += "(" + indexExpr + " == " + std::to_string(i) + " ? (" + pdfExpr + ") : ";
+      }
+
+      expr += "0.0";
+      expr += std::string(numPdfs, ')'); // Close all ternary operators
+
+      ctx.addResult(&arg, expr);
+      std::cout << "Ternary expression call used \n";
+   }
+}
+
+// RooCategory index added.
+void codegenImpl(RooCategory &arg, CodegenContext &ctx)
+{
+   int idx = ctx.observableIndexOf(&arg);
+   if (idx < 0) {
+
+      idx = 1;
+      ctx.addVecObs(arg.GetName(), idx);
+   }
+
+   // Debug output to help trace observable index mapping
+   std::cout << "[codegen debug] Category '" << arg.GetName() << "' mapped to obs[" << idx << "]\n";
+
```

Comment:
```suggestion
```
I imagine that this was useful during development, but we should not merge debugging output like this to ROOT.